packages/middleware-log-errors/lib/index.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @module @dotcom-reliability-kit/middleware-log-errors
+ */
+
+const logger = require('@financial-times/n-logger').default;
+const serializeError = require('@dotcom-reliability-kit/serialize-error');
+const serializeRequest = require('@dotcom-reliability-kit/serialize-request');
+
+/**
+ * @typedef {object} ErrorLoggingOptions
+ * @property {Array<string>} [includeHeaders]
+ *     An array of request headers to include in the log.
+ */
+
+/**
+ * Create a middleware function to log errors.
+ *
+ * @access public
+ * @param {ErrorLoggingOptions} [options = {}]
+ *     Options to configure the middleware.
+ * @returns {import('express').ErrorRequestHandler}
+ *     Returns error logging middleware.
+ */
+function createErrorLoggingMiddleware(options = {}) {
+	// Validate the included headers (this stops the request serializer from erroring
+	// on every request if the included headers are invalid)
+	const includeHeaders = options?.includeHeaders;
+	if (includeHeaders) {
+		if (
+			!Array.isArray(includeHeaders) ||
+			!includeHeaders.every((header) => typeof header === 'string')
+		) {
+			throw new TypeError(
+				'The `includeHeaders` option must be an array of strings'
+			);
+		}
+	}
+
+	return (error, request, response, next) => {
+		// We add a paranoid try/catch here because it'd be really embarassing
+		// if the error logging middleware threw an unhandled error, wouldn't it
+		try {
+			const app = {
+				name: response.getHeader('ft-app-name') || null,
```

Comment:
Would it be more reliable to read this from `process.env.SYSTEM_CODE` which Cloud Enablement requires and Tool Kit sets?

This response header _should_ get set by n-express but I don't know every case in which it might not exist. Whereas I'm pretty sure that the environment variable will be there 99.99% of the time.

Reply:
Oh that's a good idea 👍  how about something like this so we have a high likelihood of getting _something_?

```js
process.env.SYSTEM_CODE || response.getHeader('ft-app-name') || null,
```

Do you think I should do it in this PR or a separate smaller feature PR?

Reply:
Let's go new PR, then it's less of an issue if it's not merged before I go because the current way _should_ at least work for debug-user